### PR TITLE
feat(extension): add ada handle name to receive QR drawer

### PR DESCRIPTION
**Pull request by shawnbusuttil**
# Checklist

- [ ] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

This PR aims to add the handle name (if found) to the receive drawer.

## Testing

Simply click the receive button in the wallet. Note: You must have a minted ADA Handle NFT in your wallet.

## Screenshots

TBD

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/376/5256337899/index.html) for [e915e9a5](https://github.com/input-output-hk/lace/pull/41/commits/e915e9a5f98fce82c22d2f956de5362b1c6cb689)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 19     | 0      | 0       | 0     | 19    | ✅     |

### CODE REVIEW

File: apps/browser-extension-wallet/manifest.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "permissions": ["webRequest", "storage", "tabs", "unlimitedStorage"],
   "host_permissions": ["<all_urls>"],
   "content_security_policy": {
-    "extension_pages": "default-src 'self' $LOCALHOST_DEFAULT_SRC; frame-src https://connect.trezor.io/; script-src 'self' 'wasm-unsafe-eval' $LOCALHOST_SCRIPT_SRC; font-src 'self' https://use.typekit.net; object-src 'self'; connect-src $CARDANO_SERVICES_URLS https://api.coingecko.com https://analyticsv2.muesliswap.com $LOCALHOST_CONNECT_SRC https://matomo.cw.iog.io/ https://use.typekit.net; style-src * 'unsafe-inline'; img-src * data:;"
+    "extension_pages": "default-src 'self' $LOCALHOST_DEFAULT_SRC; frame-src https://connect.trezor.io/; script-src 'self' 'wasm-unsafe-eval' $LOCALHOST_SCRIPT_SRC; font-src 'self' https://use.typekit.net; object-src 'self'; connect-src $CARDANO_SERVICES_URLS https://preprod.api.handle.me https://api.coingecko.com https://analyticsv2.muesliswap.com $LOCALHOST_CONNECT_SRC https://matomo.cw.iog.io/ https://use.typekit.net; style-src * 'unsafe-inline'; img-src * data:;"
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/manifest.json:22]: `$ADA_HANDLE_URLS`

**Review by rhyslbw — CHANGES_REQUESTED** [review]
There's also a lint warning to address still.
Screenshot 2023-06-06 at 12 12 48

**Review by dmitrii-gaico — APPROVED** [review]

**Review by gabriela-ponce — APPROVED** [review]

File: apps/browser-extension-wallet/src/hooks/useGetHandles.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useWalletStore } from '@src/stores';
+import { isAdaHandleEnabled } from '@src/features/ada-handle/config';
+import { useObservable } from './useObservable';
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useGetHandles = () => {
```
Comment (greatertomi) [inline apps/browser-extension-wallet/src/hooks/useGetHandles.ts:6]: Maybe we should add a return type and remove the disable eslint lint comment
```suggestion
export const useGetHandles = (): HandleInfo[] | undefined => {
```

File: apps/browser-extension-wallet/src/hooks/useGetHandles.ts
Comment (greatertomi) [inline apps/browser-extension-wallet/src/hooks/useGetHandles.ts:1]: We can also add this hook to the hooks index file `apps/browser-extension-wallet/src/hooks/index.ts`

File: apps/browser-extension-wallet/src/hooks/useGetHandles.ts (hunk 2 above)
Comment (greatertomi) [inline apps/browser-extension-wallet/src/hooks/useGetHandles.ts:6]: Maybe we should add a return type and remove the disable eslint lint comment

```suggestion
export const useGetHandles = (): HandleInfo[] | undefined => {
```

**Review by greatertomi — CHANGES_REQUESTED** [review]

**Review by rhyslbw — APPROVED** [review]

**Review by greatertomi — APPROVED** [review]

**Pull request merged**